core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -308,6 +308,20 @@ sealed abstract class IO[+E, +A] { self =>
   final def ensuring(finalizer: IO[Nothing, Unit]): IO[E, A] =
     new IO.Ensuring(self, finalizer)
 
+  /**
+   * Executes the action on the specified `ExecutionContext` and then shifts back
+   * to the default one.
+   */
+  final def on(ec: ExecutionContext): IO[E, A] =
+    IO.shift(ec) *> this <* IO.shift
```

Comment:
I'm not as familiar with ZIO's error model yet, but could this an operation we may want to protect with `bracket`?